library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraphIT.java

```diff
@@ -241,7 +206,7 @@ public void shouldAddAndGetVertexWithNullViewAndVertexList() {
 
         // Then
         final GafferPopVertex vertex = vertices.next();
-        assertFalse(vertices.hasNext()); // there is only 1 vertex
+        assertThat(vertices).isExhausted(); // there is only 1 vertex
```

Comment:
Would something like this be maybe better:
```java
assertThat(vertices).toIterable().hasSize(1);
```

Reply:
The `extracting()` method should allow for the last case e.g.

```java
assertThat(vertices)
                .toIterable()
                .hasSize(1)
                .first()
                .hasFieldOrPropertyWithValue("id", VERTEX_1)
                .hasFieldOrPropertyWithValue("label", SOFTWARE_NAME_GROUP)
                .extracting(v -> v.property(NAME_PROPERTY).value())
                .isEqualTo("GafferPop");
```

Reply:
I've added this and included it as its own method to reduce duplication.